miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -266,16 +267,12 @@ private Sample save(Sample sample) throws IOException {
         created.setAlias(generatedAlias);
         needsUpdate = true;
       }
-      if (isStringBlankOrNull(sample.getAlias())) {
-        sample.setAlias(sample.getName());
+      if (isStringBlankOrNull(created.getAlias())) {
+        created.setAlias(created.getName());
```

Comment:
There shouldn't be a fail-safe here. The alias should either be validated as given or generated using the name generator. Above, this method also does validation *after* saving instead of before